### PR TITLE
[Typescript] Introduce Extract Override Settings method

### PR DESCRIPTION
**Pull request by LastMileNewGrad**
[Typescript] Introduce Extract Override Settings method

This diff introduces the `extractOverrideSettings` function. One of the abstractions which will make it easier to construct model sttings

- Abstracted existing logic to get overrides
- Added test. Same test & logic as the python side

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/52).
* #91
* #76
* #75
* #55
* __->__ #52

### CODE REVIEW

File: typescript/__tests__/testProgramaticallyCreateConfig.ts [hunk 1]
```diff
@@ -18,3 +20,69 @@ describe("test extract_override_settings", () => {
     expect(globalSettingsForTestModel).toEqual({ topP: 0.9 });
   });
 });
+
+describe("test extract_override_settings", () => {
```
Comment (saqadri) [inline typescript/__tests__/testProgramaticallyCreateConfig.ts:24]: Update `describe and `test` as I explained in the diff below

File: typescript/lib/util/configUtils.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *    This function takes the inference settings and a model ID and returns a subset
+ *    of inference settings that have been overridden by model-specific settings. It
+ *    compares the provided settings with global settings, and returns only those that
+ *    differ or have no corresponding global setting.
+ * @param configRuntime The AIConfigRuntime that the prompt belongs to.
+ * @param inferenceSettings The model settings from the input data.
+ * @param modelId The model ID of the prompt.
+ * @returns The model settings from the input data.
+ */
+export function extractOverrideSettings(
```
Comment (saqadri) [inline typescript/lib/util/configUtils.ts:18]: Just put this in `utils.ts`

File: typescript/lib/util/configUtils.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  configRuntime: AIConfigRuntime,
+  inferenceSettings: InferenceSettings,
+  modelId: string
```
Comment (saqadri) [inline typescript/lib/util/configUtils.ts:21]: `modelName` since that's what we call it everywhere

File: typescript/lib/util/configUtils.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+) {
+  let modelMetadata: ModelMetadata | string;
+  const modelName = modelId;
+  const globalModelSettings: InferenceSettings =
+    configRuntime.getGlobalSettings(modelName) ?? ({} as InferenceSettings);
```
Comment (saqadri) [inline typescript/lib/util/configUtils.ts:26]: `const globalModelSettings = { ...(configRuntime.getGlobalSettings(modelName) ?? {} }`

Same for inferenceSettings below.

**Review by saqadri — APPROVED** [review]
Approving to unblock but some comments

File: typescript/__tests__/testProgramaticallyCreateConfig.ts (hunk 1 above)
Comment (LastMileNewGrad) [reply typescript/__tests__/testProgramaticallyCreateConfig.ts]: Updated. Please take another look

File: typescript/lib/util/configUtils.ts (hunk 2 above)
Comment (LastMileNewGrad) [reply typescript/lib/util/configUtils.ts]: moved to utils.ts

File: typescript/lib/util/configUtils.ts (hunk 3 above)
Comment (LastMileNewGrad) [reply typescript/lib/util/configUtils.ts]: updated

File: typescript/lib/util/configUtils.ts (hunk 4 above)
Comment (LastMileNewGrad) [reply typescript/lib/util/configUtils.ts]: updated.

File: typescript/__tests__/testProgramaticallyCreateConfig.ts (hunk 1 above)
Comment (saqadri) [reply typescript/__tests__/testProgramaticallyCreateConfig.ts]: Looks good, ship it

**Pull request merged**